### PR TITLE
enforce usage of vmi_lock_guard RAII pattern

### PR DESCRIPTION
**Pull request by disaykin**

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/plugins/plugins.h [hunk 1]
```diff
@@ -392,11 +366,13 @@ struct vmi_lock_guard
...
     {
-        unlock();
+        drakvuf_release_vmi(drakvuf);
     }
 
+private:
     drakvuf_t drakvuf;
-    vmi_instance_t vmi;
+public:
+    vmi_instance_t vmi; // TODO: make field private
```
Comment (tklengyel) [inline src/plugins/plugins.h:375]: Why doesn't it work as private as-is?

Comment (disaykin) [reply src/plugins/plugins.h:375]: Some code actively accesses this field as public member. I weel need time to fix this in the next step.

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Pull request closed without merging**